src/drivers/serial/am85c30/am85c30.c

```diff
@@ -8,32 +6,34 @@
  * @date 18.12.2017
  */
 
+#include <stdint.h>
+
+#include "../am85c30/am85c30.h"
+
 #include <asm/io.h>
 #include <drivers/common/memory.h>
 #include <drivers/diag.h>
 #include <drivers/serial/uart_device.h>
 #include <drivers/serial/diag_serial.h>
 #include <embox/unit.h>
 
-
 EMBOX_UNIT_INIT(uart_init);
 
-#define UART_BASE   0x83200040
+#define UART_BASE   (uintptr_t)0x83200040
 #define	IRQ_NUM     3 /* ZILOG_IRQ_DEFAULT 3 */ /* default IRQ # */
 
-
 static inline void
 am85c30_com_outb(void *iomem_addr, uint8_t data)
 {
-	e2k_write8(data, UART_BASE + 1);
+	e2k_write8(data, (void*)(UART_BASE + 1));
```

Comment:
May be it's a better idea to change `e2k_readX/e2k_writeX` functions to use second argument as uintptr_t? Otherwise we'll have to convert it to (void*) everywhere.

Reply:
Done